### PR TITLE
Toggle preprocess btn

### PR DESCRIPTION
**Pull request by rnugraha**

### CODE REVIEW

**Comment by sherzinger** [comment]
looks good except of your piping of "subsets".
Wouldn't it be far easier to make the checking directly via "isSubsetEmpty(2)"?

**Comment by rnugraha** [comment]
In which line?

File: web-app/js/smartR/_angular/templates/sortingCriteria.html [hunk 1]
```diff
@@ -3,43 +3,43 @@
...
+            <input type="radio" ng-model="criteria"  value="mean"> Mean
         </label></div>
         <div><label>
-            <input type="radio" ng-model="criteria" name="rankCriteria" value="median"> Median
+            <input type="radio" ng-model="criteria"  value="median"> Median
         </label></div>
     </fieldset>
 </div>
-<div class="heim-input-field-sub" id="sr-multi-subset">
+<div class="heim-input-field-sub" id="sr-multi-subset" ng-if="subsets > 1">
```
Comment (sherzinger) [inline web-app/js/smartR/_angular/templates/sortingCriteria.html:26]: and here

File: web-app/js/smartR/_angular/services/processService.js [hunk 2]
```diff
@@ -37,6 +58,16 @@ window.smartRApp.factory('processService', [ function() {
             _toggleButton(CAPTURE_BTN, true);
             _toggleButton(DOWNLOAD_BTN, true);
         }
+        if (!value && status === service.status.SUCCESS) {
+            var _sumData = service.components[FETCH_BTN].summaryData;
+            // toggle preprocess button based on number of fetched samples
+            // disable it when sample only one.
+            _toggleButton(PREPROCESS_BTN, _sumData.allSamples <= 1);
+            // toggle rank criteria based on number of fetched samples and number of subsets
+            _toggleRankCriteria(SORTING_CRITERIA, _sumData.allSamples, _sumData.subsets);
```
Comment (sherzinger) [inline web-app/js/smartR/_angular/services/processService.js:67]: and here

File: web-app/js/smartR/_angular/directives/sortingCriteria.js [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-'use strict';
-
-window.smartRApp.directive('sortingCriteria', ['$rootScope', function($rootScope) {
+window.smartRApp.directive('sortingCriteria', ['$rootScope', 'processService', function($rootScope, processService) {
     return {
         restrict: 'E',
         scope: {
-            criteria : '='
+            criteria : '=',
+            subsets : '='
```
Comment (sherzinger) [inline web-app/js/smartR/_angular/directives/sortingCriteria.js:8]: and here

File: web-app/js/smartR/_angular/controllers/heatmap.js [hunk 4]
```diff
@@ -37,6 +37,7 @@ window.smartRApp.controller('HeatmapController',
                 sorting : 'nodes',
                 ranking : 'coef'
             },
+            subsets : 0,
```
Comment (sherzinger) [inline web-app/js/smartR/_angular/controllers/heatmap.js:40]: and here

File: grails-app/views/layouts/_heatmap.gsp [hunk 5]
```diff
@@ -83,7 +83,8 @@
 
             %{--Type of sorting to apply--}%
             <div class="heim-input-field  sr-input-area">
-                <sorting-criteria criteria="runAnalysis.params.ranking"></sorting-criteria>
+                <sorting-criteria criteria="runAnalysis.params.ranking" subsets="runAnalysis.subsets">
```
Comment (sherzinger) [inline grails-app/views/layouts/_heatmap.gsp:86]: here

**Comment by rnugraha** [comment]
Oh that function. Actually I try not to use it .. that's a nasty dependency. It's from RModules file even. We actually should start to decouple those dependencies even to javascript from transmartApp if we want to make smartR loosly coupled. 

Maybe by re-creating it in this module. I'll come up with something.

**Comment by sherzinger** [comment]
Just put it in SmartR Utils maybe

**Comment by rnugraha** [comment]
The original isSubsetEmpty is even not for checking if resultInstanceIds already created. 

It only checks if there's node dropped in the associated cohort boxes. 
so if you do isSubsetEmpty(2) .. meaning u are asking if subset 2 is empty or not. This could also means subset 2 is not empty but actually there is only one result instance id.

What we need here whether is there one or two subsets has been created (reflected by created rsID) .. not only checking if the drop zones are empty or not.

So in short, I will stick to this approach. It's still more legit to calculate the number of subsets after fetching the data. In this way, we are sure how many result instance ids has been created
